File: daq/backend/app/models/dataset.py
from datetime import datetime

from pydantic import BaseModel


class DatasetRecord(BaseModel):
    slug: str
    filename: str
    original_name: str
    title: str
    uploaded_at: datetime
    size_bytes: int


class DatasetListItem(BaseModel):
    slug: str
    title: str
    filename: str
    uploaded_at: datetime
    size_bytes: int

File: daq/frontend/src/types/dataset.ts
export type Dataset = {
  slug: string;
  title: string;
  filename: string;
  uploaded_at: string;
  size_bytes: number;
};

export type SchemaColumn = {
  name: string;
  type: string;
  unit?: string | null;
  display_name?: string;
  sample_values: string[];
};
